transcode/transcode.go

```diff
@@ -507,37 +511,53 @@ func transcodeSegment(
 ) error {
 	start := time.Now()
 
-	transcodeConf := clients.LivepeerTranscodeConfiguration{
-		TimeoutMultiplier: 10,
-	}
-	// If this is a request to transcode a Clip source input, then
-	// force T to do a re-init of transcoder after segment at idx=0.
-	// This is required because the segment at idx=0 is a locally
-	// re-encoded segment and the following segment at idx=1 is a
-	// source recorded segment. Without a re-init of the transcoder,
-	// the different encoding between the two segments causes the
-	// transcode operation to incorrectly tag the output segment as
-	// having two video tracks.
-	if transcodeRequest.IsClip && (int64(segment.Index) == 0 || segment.IsLastSegment) {
-		transcodeConf.ForceSessionReinit = true
-	} else {
-		transcodeConf.ForceSessionReinit = false
-	}
 	// This is a temporary workaround that implements the same logic
 	// as the previous if block -- a new manifestID will force a
 	// T session re-init between segment at index=0 and index=1.
 	if transcodeRequest.IsClip && (int64(segment.Index) == 0 || segment.IsLastSegment) {
 		manifestID = manifestID + "_clip"
 	}
 
+	transcodeProfiles := make([]video.EncodedProfile, 0, len(encodedProfiles))
+	copySource := false
+	for _, profile := range encodedProfiles {
+		if profile.Copy {
+			if copySource {
+				return fmt.Errorf("multiple source copy profiles found")
+			}
+			copySource = true
+		} else {
+			transcodeProfiles = append(transcodeProfiles, profile)
+		}
+	}
+
 	var tr clients.TranscodeResult
+	var sourceSegment *bytes.Buffer
 	err := backoff.Retry(func() error {
 		ctx, cancel := context.WithTimeout(context.Background(), clients.MaxCopyFileDuration)
 		defer cancel()
 		rc, err := clients.GetFile(ctx, transcodeRequest.RequestID, segment.Input.URL.String(), nil)
 		if err != nil {
 			return fmt.Errorf("failed to download source segment %q: %s", segment.Input, err)
 		}
+		defer rc.Close()
+
+		var r io.Reader = rc
```

Comment:
Maybe better to stick this block into a separate function so we can unit test it?

Reply:
I think it's a little overkill since this is only 2 ifs, the complex/testable logic is inside `io.TeeReader` and `io.Copy` IMO. I definitely have a bias for too few tests tho, so I'll follow your suggestion anyway!